src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringAfterJoin = '';
+    const finallySeparator = separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        stringAfterJoin += this[i];
+      }
+
+      if (i < this.length - 1 && finallySeparator !== undefined) {
```

Comment:
You set a default value for the separator, so it will never be equal to `undefined`